lib/charms/opensearch/v0/opensearch_relation_provider.py

```diff
@@ -332,15 +332,16 @@ def update_certs(self, relation_id, ca_chain=None):
             return
         if not ca_chain:
             try:
-                ca_chain = self.charm.secrets.get_object(Scope.APP, CertType.APP_ADMIN.val).get(
-                    "chain"
-                )
+                if ca_chain := self.charm.secrets.get_object(
+                    Scope.APP, CertType.APP_ADMIN.val
+                ).get("chain"):
+                    self.opensearch_provides.set_tls_ca(relation_id, ca_chain)
+                else:
+                    logger.warning("unable to get ca_chain")
```

Comment:
If I get it right, the `if ... else ...` can't raise `AttributeError`.

Why not to move it after the `try ... except`, based on whether `ca_chain` is defeined or not?

Reply:
Hi @juditnovak the `AttributeError` is happening because the original `ca_chain` is set to `None` at early stages.

Reply:
Exactly.

```
           try:
                ca_chain := self.charm.secrets.get_object(
                    Scope.APP, CertType.APP_ADMIN.val
                ).get("chain")
            except AttributeError:
                # cert doesn't exist - presumably we don't yet have a TLS relation.
                return
                
          if ca_chain:
               self.opensearch_provides.set_tls_ca(relation_id, ca_chain)
           else:
                logger.warning("unable to get ca_chain")
```
Generally we should have as little in the `try ... except` as possible